### ISSUE TITLE
SocketTaskExtensions Class is not supported on .Net Framework 4.6.2

### ISSUE THREAD

**Issue by sandra-dcs** [robinrodricks/FluentFTP#1054]
FTP Server OS: Windows 

FTP Server Type: Windows Server IIS 

Client Computer OS: Windows 

FluentFTP Version: 4.2 

Framework: .NET  4.6.2

.Net 4.6.2 does not support SocketTaskExtensions so it fails to compile.

The problem is in Streams\FtpSocketStream.cs

The following method fixes.

```
	/// <summary>
		/// Connect to the specified host
		/// Detects timeout and throws that explicitly
		/// </summary>
		/// <param name="ipad">The ip address to connect to</param>
		/// <param name="port">The port to connect to</param>
		private async Task<bool> ConnectAsyncHelper(IPAddress ipad, int port, CancellationToken token) {

			int ctmo = this.ConnectTimeout;

#if NETSTANDARD
			try {
				using (var timeoutSrc = CancellationTokenSource.CreateLinkedTokenSource(token)) {
					timeoutSrc.CancelAfter(ctmo);
					await EnableCancellation(m_socket.ConnectAsync(ipad, port), timeoutSrc.Token, () => DisposeSocket());
				}
			}
			catch (SocketException ex) {
				if (ex.SocketErrorCode == SocketError.OperationAborted ||
					ex.SocketErrorCode == SocketError.TimedOut) {
					throw new TimeoutException("Timed out trying to connect!");
				}
				throw;
			}
#else
			 try {
#if v472
				 
				    using (var timeoutSrc = CancellationTokenSource.CreateLinkedTokenSource(token)) {
					timeoutSrc.CancelAfter(ctmo);

                    await EnableCancellation(m_socket.ConnectAsync(ipad, port), timeoutSrc.Token, () => DisposeSocket());
					}
#else
				using (var timeoutSrc = CancellationTokenSource.CreateLinkedTokenSource(token)) {
					timeoutSrc.CancelAfter(ctmo);
					var connectResult = m_socket.BeginConnect(ipad, port, null, null);
					await EnableCancellation(Task.Factory.FromAsync(connectResult, m_socket.EndConnect), timeoutSrc.Token, () => DisposeSocket());
				}
#endif
			}
			catch (SocketException ex) {
				if (ex.SocketErrorCode == SocketError.OperationAborted ||
					ex.SocketErrorCode == SocketError.TimedOut) {
					throw new TimeoutException("Timed out trying to connect!");
				}
				throw;
			}
			catch (ObjectDisposedException ex) {
				throw new TimeoutException("Timed out trying to connect!");
			}
#endif
					return m_socket.Connected;
		}
```

**Comment by robinrodricks** [comment]
Why are you compiling from source? Just use the nuget?

**Comment by robinrodricks** [comment]
Also, we are compiling natively on all these and it works, I just published a nuget.  No idea what this error is.

![image](https://user-images.githubusercontent.com/6797866/202851077-f88f2142-0d06-4755-bbb8-d195edb60836.png)

**Comment by FanDjango** [comment]
@sandra-dcs Please see PR #1056.

Question: Do you believe a change to `MoveFile(...)` like yours, is an enhancement we should include?

**Comment by sandra-dcs** [comment]
I think it will be a very useful enhancement. I
I download files from a server, and once downloaded they are moved to an archive folder. The archive folder contains 1000's of files, so it saves time not having to perform a file listing each time.
It will save time when uploading files too, I upload to temp folder and then move them to a folder where they are processed, so when transmitting 100's of files it will make a big difference.
I work on the assumption only delete the file if the move fails.

**Comment by robinrodricks** [comment]
I think we can add it @FanDjango  it seems good. @sandra-dcs  can you file a PR please? We can take it forward...

**Comment by FanDjango** [comment]
I have a PR ready, I'll put it in and you guys can discuss...

**Comment by FanDjango** [comment]
Released as V42.1.0
